src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -96,16 +93,16 @@ public void OnSourceFileUpdated(DocumentId documentId)
 
         public void StartDebuggingSession()
         {
-            var previousSession = Interlocked.CompareExchange(ref _debuggingSession, new DebuggingSession(_workspace, _debugeeModuleMetadataProvider, _activeStatementProvider, _compilationOutputsProvider), null);
+            var previousSession = Interlocked.CompareExchange(ref _debuggingSession, new DebuggingSession(_workspace, _debugeeModuleMetadataProvider, _compilationOutputsProvider), null);
             Contract.ThrowIfFalse(previousSession == null, "New debugging session can't be started until the existing one has ended.");
         }
 
-        public void StartEditSession()
+        public void StartEditSession(ActiveStatementProvider activeStatementsProvider)
```

Comment:
The debugger now passes a provider to each edit session, rather than having a global one. 
The reason being that active statements don't have any meaning outside of edit session (break mode). 
The provider is just a get method. I could define an abstract class/interface with just single method and a derived type that implements it, but a delegate is simpler and allows the code to be more succinct as you can just pass in a lambda.

Reply:
>  Does it change across edit sessions?

Yes. It is scoped to a session.